### PR TITLE
Add screen sharing to SDK

### PR DESCRIPTION
**Pull request by DanielNovak**
Add screensharing ability to the Video SDK.

### CODE REVIEW

File: dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt [hunk 1]
```diff
@@ -54,6 +59,21 @@ internal fun SettingsMenu(
...
+        contract = ActivityResultContracts.StartActivityForResult(),
+        onResult = {
+            if (it.resultCode == Activity.RESULT_OK && it.data != null) {
+                call.startScreenSharing(it.data!!)
+            }
+            onDismissed.invoke()
+        },
+    )
+
+    val isScreenSharing = call.screenShare.isEnabled.collectAsState()
```
Comment (skydoves) [inline dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt]: We can observe it using delegation.

```kotlin
val isScreenSharing by call.screenShare.isEnabled.collectAsState()
```

File: dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt [hunk 2]
```diff
@@ -92,6 +112,37 @@ internal fun SettingsMenu(
 
                 Spacer(modifier = Modifier.height(12.dp))
 
+                Row(
+                    modifier = Modifier.clickable {
+                        if (!isScreenSharing.value) {
```
Comment (skydoves) [inline dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt]: With the delegation observation, we can tweak this:

```kotlin
if (!isScreenSharing) {
```

File: dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt [hunk 3]
```diff
@@ -54,6 +59,21 @@ internal fun SettingsMenu(
...
 
+    val screenSharePermissionResult = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.StartActivityForResult(),
+        onResult = {
+            if (it.resultCode == Activity.RESULT_OK && it.data != null) {
+                call.startScreenSharing(it.data!!)
+            }
+            onDismissed.invoke()
+        },
+    )
```
Comment (skydoves) [inline dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt:71]: I think we can make this simpler by building dedicated Compose APIs. Let me create one once we merge this PR.

**Review by skydoves — COMMENTED** [review]
Overall looks good to me! I left several trivial comments about Compose.

File: dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt (hunk 3 above)
Comment (DanielNovak) [reply dogfooding/src/main/kotlin/io/getstream/video/android/ui/call/SettingsMenu.kt:71]: Nice. Yes - I was thinking if we maybe can help a bit more with the integration if we provide something in the SDK to handle this permission.

**Review by skydoves — APPROVED** [review]

**Pull request merged**